tests/test_enterprise/api/test_serializers.py

```diff
@@ -628,3 +631,74 @@ def test_serialize_users(self):
         serializer = EnterpriseMembersSerializer(serializer_input_2)
         serialized_user = serializer.data
         self.assertEqual(serialized_user, expected_user_2)
+
+
+@mark.django_db
+class TestEnterpriseCourseEnrollmentAdminViewSerializer(TestCase):
+    """ Unit tests for EnterpriseCourseEnrollmentAdminViewSerializer. """
+
+    def setUp(self):
+        """ Set up test data. """
+        super().setUp()
+        self.user = factories.UserFactory.create(is_staff=True, is_active=True)
+        self.enterprise_customer_user = factories.EnterpriseCustomerUserFactory(
+            user_id=self.user.id,
+        )
+        self.enterprise_customer = self.enterprise_customer_user.enterprise_customer
+
+    @patch.object(HttpRequest, 'get_host', return_value='example.edx.org')
+    @patch('enterprise.models.CourseEnrollment')
+    @patch('enterprise.api.v1.serializers.get_certificate_for_user')
+    def test_serialization_basic(
+        self,
+        mock_get_certificate,
+        mock_course_enrollment_class,
+        _,
+    ):
```

Comment:
The serializer has a large number of conditionals, but I only see one dataset tested.
You may consider using the `@ddt` decorator to test this with different datasets.
Or break the serializer code into smaller functions tested on its own, which makes it easier to test edge cases.